### PR TITLE
added search mapping , liquibase , soft-delete

### PR DESCRIPTION
**Pull request by dmytrocherepov**

### CODE REVIEW

File: src/main/java/com/example/bookstoreapp/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +37,21 @@ public BookDto getById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
+
+    @PutMapping("/{id}")
```
Comment (Alexsandr-Yablunovskyi) [inline src/main/java/com/example/bookstoreapp/controller/BookController.java:41]: Maybe it's better to put @ResponseStatus (with appropriate status code) to all methods

File: src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -39,4 +43,24 @@ public BookDto getById(Long id) {
...
+    public BookDto updateBookById(CreateBookRequestDto requestDto, Long id) {
+        Book book = bookMapper.toBook(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
```
Comment (Alexsandr-Yablunovskyi) [inline src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java:57]: What happens if there is no row in the database with the requested ID?

**Review by Alexsandr-Yablunovskyi — APPROVED** [review]
Great job! Just few minor quastions/recomendations, but it's up to you

File: src/main/java/com/example/bookstoreapp/controller/BookController.java [hunk 3]
```diff
@@ -32,4 +37,21 @@ public BookDto getById(@PathVariable Long id) {
...
+    public BookDto updateBookById(
+            @RequestBody CreateBookRequestDto requestDto, @PathVariable Long id) {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/controller/BookController.java]:
```suggestion
    public BookDto updateBookById(
            @RequestBody CreateBookRequestDto requestDto, 
            @PathVariable Long id
            ) {
```

File: src/main/java/com/example/bookstoreapp/dto/BookSearchParameters.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package com.example.bookstoreapp.dto;
+
+public record BookSearchParameters(String[] authors, String[] titles) {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/dto/BookSearchParameters.java]:
```suggestion
public record BookSearchParameters(
String[] authors, 
String[] titles
) {
```

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                  name: cover_image
+                  type: VARCHAR(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
+
+
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]:
```suggestion
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: Let’s leave one empty line at the end of the file

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few minor mistakes. Please fix it before merging;)

**Pull request merged**